FirebaseFunctions/Sources/Functions.swift

```diff
@@ -471,6 +471,215 @@ enum FunctionsConstants {
+  /// - Returns: HTTPSCallableResult Streaming.
+  @available(macOS 12.0, iOS 15.0, watchOS 8.0, tvOS 15.0, *)
+  func stream(at url: URL,
+              data: Any?,
+              options: HTTPSCallableOptions?,
+              timeout: TimeInterval)
+    -> AsyncThrowingStream<JSONStreamResponse, Error> {
+    AsyncThrowingStream { continuation in
+      Task {
+        let urlRequest: URLRequest
+        do {
+          let context = try await contextProvider.context(options: options)
+          urlRequest = try makeRequestForStreamableContent(
+            url: url,
+            data: data,
+            options: options,
+            timeout: timeout,
+            context: context
+          )
+        } catch {
+          continuation.finish(throwing: FunctionsError(
+            .invalidArgument,
+            userInfo: [NSUnderlyingErrorKey: error]
+          ))
+          return
+        }
+
+        // TODO: Vertex prints the curl command. Should this?
+        //      #if DEBUG
+        //        printCURLCommand(from: urlRequest)
+        //      #endif
+
+        let stream: URLSession.AsyncBytes
+        let rawResponse: URLResponse
+        do {
+          (stream, rawResponse) = try await URLSession.shared.bytes(for: urlRequest)
+        } catch {
+          continuation.finish(throwing: FunctionsError(
+            .unavailable,
+            userInfo: [NSUnderlyingErrorKey: error]
+          ))
+          return
+        }
+
+        // Verify the status code is an HTTP response.
+        guard let response = rawResponse as? HTTPURLResponse else {
+          continuation.finish(
+            throwing: FunctionsError(
+              .unavailable,
+              userInfo: [NSLocalizedDescriptionKey: "Response was not an HTTP response."]
+            )
+          )
+          return
+        }
+
+        // Verify the status code is a 200.
+        guard response.statusCode == 200 else {
+          continuation.finish(
+            throwing: FunctionsError(
+              httpStatusCode: response.statusCode, body: nil, serializer: serializer
+            )
+          )
+          return
+        }
+
+        do {
+          for try await line in stream.lines {
+            if line.hasPrefix("data:") {
+              // We can assume 5 characters since it's utf-8 encoded, removing `data:`.
+              let jsonText = String(line.dropFirst(5))
+              let data: Data
+              do {
+                data = try jsonData(jsonText: jsonText)
+              } catch {
+                continuation.finish(throwing: error)
+                return
+              }
+
+              // Handle the content and parse it.
+              do {
+                let content = try callableStreamResult(fromResponseData: data)
+                continuation.yield(content)
+              } catch {
+                continuation.finish(throwing: error)
+                return
+              }
```

Comment:
@ncooke3 can these 2 `do` statements being collapse into one, given that the catch is the same?